plexanisync/anilist.py

```diff
@@ -92,44 +104,53 @@ def match_to_plex(self, anilist_series: List[AnilistSeries], plex_series_watched
+                            (item for item in anilist_matches if item.anilist_id == matched_id),
                             None,
                         )
 
                         if not match:
                             # Create first match dict for this anilist id
-                            anilist_matches.append({
-                                "anilist_id": matched_id,
-                                "watched_episodes": plex_season.watched_episodes - mapped_start + 1,
-                                "total_episodes": plex_season.last_episode,
-                                "mapped_seasons": [plex_season.season_number],
-                            })
+                            anilist_matches.append(AnilistMatch(
+                                matched_id,
+                                plex_season.watched_episodes - mapped_start + 1,
+                                plex_season.last_episode,
+                                [plex_season.season_number],
+                                [plex_season.rating]
+                            ))
                             continue
                         # For multiple seasons with the same id
                         # If the start of this season has been mapped use that.
                         if mapped_start != 1:
-                            match["watched_episodes"] = plex_season.watched_episodes - mapped_start + 1
+                            match.watched_episodes = plex_season.watched_episodes - mapped_start + 1
                         else:
-                            match["watched_episodes"] += plex_season.watched_episodes
+                            match.watched_episodes += plex_season.watched_episodes
 
                         # TODO support using number of last episode of the last season as a start
-                        match["mapped_seasons"].append(plex_season.season_number)
+                        match.mapped_seasons.append(plex_season.season_number)
+                        match.ratings.append(plex_season.rating)
 
                 for match in anilist_matches:
                     logger.info(
                         "Custom Mapping of Title found | "
-                        f"title: {plex_title} | anilist id: {match['anilist_id']} | "
-                        f"total watched episodes: {match['watched_episodes']} | "
-                        f"seasons with the same anilist id: {match['mapped_seasons']}"
+                        f"title: {plex_title} | anilist id: {match.anilist_id} | "
+                        f"total watched episodes: {match.watched_episodes} | "
+                        f"seasons with the same anilist id: {match.mapped_seasons}"
                     )
 
+                    season_ratings = match.ratings
+                    # filter out unrated seasons
+                    season_ratings = [r for r in season_ratings if r != 0]
```

Comment:
The list comprehension will return empty list so it's not necessary to assign to season_ratings first right?

Reply:
You're right, I changed it